sound/soc/sof/intel/hda-pcm.c

```diff
@@ -136,6 +137,38 @@ int hda_dsp_pcm_trigger(struct snd_sof_dev *sdev,
 	return hda_dsp_stream_trigger(sdev, stream, cmd);
 }
 
+snd_pcm_uframes_t hda_dsp_pcm_pointer(struct snd_sof_dev *sdev,
+				      struct snd_pcm_substream *substream)
+{
+	struct hdac_stream *hstream = substream->runtime->private_data;
+	struct hdac_ext_stream *stream = stream_to_hdac_ext_stream(hstream);
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	snd_pcm_uframes_t pos = 0;
+
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
```

Comment:
please comment on logic here and why there is difference for playback and capture.